Update arrays.rst

StephenVavasis opened the pull request:
Remark about sub performance

StephenVavasis committed:
Update arrays.rst

Remark about sub performance

omus commented on doc/manual/arrays.rst:
```diff
@@ -475,6 +475,14 @@ the same as the result of ``getindex``, except the data is left in place.
+achieved by explicitly
+copying subarrays into and out of temporary storage with for-loops rather than
+invoking ``sub``.
+
```
Should redo the line wrapping

StephenVavasis committed:
Update arrays.rst

Made the line-wrapping more even and also reworded the paragraph for improved clarity

mbauman commented:
Sorry this fell through the cracks!  I think it could still be worth stating.  Do we have any notes like it anywhere else?

StephenVavasis commented:
Just wondering... instead of a `for` loop, can we now recommend a dot operation for very small subvectors?   For example, does the following code entail any heap allocation?

```Julia
   u = zeros(2)
    for k = 1 : n  # main loop here
        u[:] .= v[k : k + 1]  # dot op instead of a for-loop
        my_operation!(u)
        v[k : k + 1] .= u  # instead of a for-loop
    end
```

KristofferC commented:
I think it will allocate a `view`.

laborg commented:
The statement in this PR isn't true anymore after 1.5 because views no longer forces allocation.
